### DISCUSSION TITLE
Unable to received verification code via email

### DISCUSSION THREAD

**Discussion by steveharman** [plausible/analytics#809] [Self-Hosted Support] [answered]
## Prerequisites
- [X ] I have searched open and closed issues to make sure that the bug has not yet been reported.

## Bug report
**Describe the bug**
I am unable to receive a verification code via email

**Expected behavior**

**Screenshots (If applicable)**
Receive an email containing the verification code

**Environment (If applicable):**
 - Ubuntu 18.0.4

•The installation went fine via Docker, I can browse to the install on 8000 and enter my credentials.  
• However there seems to be an issue sending the verification code to me.  I host my email domain with Migadu.com if that's relevant?  
• The email address works fine - it received the confirmation emails from Plausible when I registered via plausible.io.

```
mail_1                 |   239 LOG: MAIN
mail_1                 |   239   <= hello@plausible.local H=hosting_plausible_1.hosting_default (MDOMAIN.com) [172.18.0.5] P=esmtp S=1025
mail_1                 |   239 LOG: smtp_connection MAIN
mail_1                 |   239   SMTP connection from hosting_plausible_1.hosting_default (MYDOMAIN.com) [172.18.0.5] closed by QUIT
mail_1                 |   240 Exim version 4.89 uid=101 gid=101 pid=240 D=40001
mail_1                 | Berkeley DB: Berkeley DB 5.3.28: (September  9, 2013)
mail_1                 | Support for: crypteq iconv() IPv6 GnuTLS move_frozen_messages DKIM DNSSEC Event OCSP PRDR SOCKS TCP_Fast_Open
mail_1                 | Lookups (built-in): lsearch wildlsearch nwildlsearch iplsearch cdb dbm dbmjz dbmnz dnsdb dsearch nis nis0 passwd
mail_1                 | Authenticators: cram_md5 plaintext
mail_1                 | Routers: accept dnslookup ipliteral manualroute queryprogram redirect
mail_1                 | Transports: appendfile/maildir/mailstore autoreply lmtp pipe smtp
mail_1                 | Fixed never_users: 0
mail_1                 | Configure owner: 0:0
mail_1                 | Size of off_t: 8
mail_1                 |   240 delivering 1lJHEw-00003r-53
mail_1                 |   240 R: dnslookup for MYEMAIL@MYDOMAIN.com (I have a valid email address! in my plausible-conf.env!)
mail_1                 |   241 T: remote_smtp for MYEMAIL@MYDOMAIN.com
mail_1                 |   241 Connecting to aspmx1.migadu.com [2001:41d0:2:4a6f::]:25 ... failed: Cannot assign requested address
mail_1                 |   241 LOG: MAIN
mail_1                 |   241   H=aspmx1.migadu.com [2001:41d0:2:4a6f::] Cannot assign requested address
mail_1                 |   241 Connecting to aspmx1.migadu.com [2001:41d0:8:6d80::]:25 ... failed: Cannot assign requested address
mail_1                 |   241 LOG: MAIN
mail_1                 |   241   H=aspmx1.migadu.com [2001:41d0:8:6d80::] Cannot assign requested address
mail_1                 |   241 Connecting to aspmx1.migadu.com [2001:41d0:2:bcc0::]:25 ... failed: Cannot assign requested address
mail_1                 |   241 LOG: MAIN
mail_1                 |   241   H=aspmx1.migadu.com [2001:41d0:2:bcc0::] Cannot assign requested address
```

Earlier in the docker-compuse up process I also have these warnings:

```
plausible_events_db_1  | Processing configuration file '/etc/clickhouse-server/config.xml'.
plausible_events_db_1  | Merging configuration file '/etc/clickhouse-server/config.d/docker_related_config.xml'.
plausible_events_db_1  | Merging configuration file '/etc/clickhouse-server/config.d/logging.xml'.
mail_1                 |     1   exim 4.89 daemon started: pid=1, -q30m, listening for SMTP on port 25 (IPv6 and IPv4)
plausible_events_db_1  | Logging warning to /var/log/clickhouse-server/clickhouse-server.log
plausible_events_db_1  | Logging errors to /var/log/clickhouse-server/clickhouse-server.err.log
plausible_events_db_1  | Logging warning to console
plausible_events_db_1  | 2021.03.08 15:03:07.565488 [ 46 ] {} <Warning> Application: Listen [::]:8123 failed: Poco::Exception. Code: 1000, e.code() = 0, e.displayText() = DNS error: EAI: Address family for hostname not supported (version 21.2.5.5 (official build)). If it is an IPv6 or IPv4 address and your host has disabled IPv6 or IPv4, then consider to specify not disabled IPv4 or IPv6 address to listen in <listen_host> element of configuration file. Example for disabled IPv6: <listen_host>0.0.0.0</listen_host> . Example for disabled IPv4: <listen_host>::</listen_host>
plausible_events_db_1  | 2021.03.08 15:03:07.567220 [ 46 ] {} <Warning> Application: Listen [::]:9000 failed: Poco::Exception. Code: 1000, e.code() = 0, e.displayText() = DNS error: EAI: Address family for hostname not supported (version 21.2.5.5 (official build)). If it is an IPv6 or IPv4 address and your host has disabled IPv6 or IPv4, then consider to specify not disabled IPv4 or IPv6 address to listen in <listen_host> element of configuration file. Example for disabled IPv6: <listen_host>0.0.0.0</listen_host> . Example for disabled IPv4: <listen_host>::</listen_host>
plausible_events_db_1  | 2021.03.08 15:03:07.567719 [ 46 ] {} <Warning> Application: Listen [::]:9009 failed: Poco::Exception. Code: 1000, e.code() = 0, e.displayText() = DNS error: EAI: Address family for hostname not supported (version 21.2.5.5 (official build)). If it is an IPv6 or IPv4 address and your host has disabled IPv6 or IPv4, then consider to specify not disabled IPv4 or IPv6 address to listen in <listen_host> element of configuration file. Example for disabled IPv6: <listen_host>0.0.0.0</listen_host> . Example for disabled IPv4: <listen_host>::</listen_host>
plausible_events_db_1  | 2021.03.08 15:03:07.568194 [ 46 ] {} <Warning> Application: Listen [::]:9004 failed: Poco::Exception. Code: 1000, e.code() = 0, e.displayText() = DNS error: EAI: Address family for hostname not supported (version 21.2.5.5 (official build)). If it is an IPv6 or IPv4 address and your host has disabled IPv6 or IPv4, then consider to specify not disabled IPv4 or IPv6 address to listen in <listen_host> element of configuration file. Example for disabled IPv6: <listen_host>0.0.0.0</listen_host> . Example for disabled IPv4: <listen_host>::</listen_host>
```

**Comment by Philipinho** [comment, marked as answer]
It seems your server is not configured to send out 
emails.
Did you setup the SMTP?

In the meantime, you can bypass the email verification using
`docker exec hosting_plausible_db_1 psql -U postgres -d plausible_db -c "UPDATE users  SET email_verified = true;"`

**Reply by steveharman** [reply]
Hi!

I didn't see any steps in the instructions that said I needed to configure SMTP. ;-). 

> "The only thing you need to install Plausible Analytics is a server with Docker installed. The server must have a CPU with x86_64 architecture and support for SSE 4.2 "  

https://plausible.io/docs/self-hosting

I don't really want the server where I've installed Plasuible to have SMTP configured tbh - if I'd read that requirement, I perhaps wouldn't have gone ahead - it didn't cross my mind that an analytics tool would need to send me emails.

Which brings me on to a related thing now I've got this far.  I've installed Plausible on a db server away from the webserver.  The db server does not have a host entry in dns and doesn't form part of a domain.  Is this going to be an issue for the web server / content that has Plausible snippet embedded?

I notice in the Plausible Wordpress plugin it states:

> "If you're self-hosting Plausible on your own infrastructure, enter the domain name where you installed it to enable the integration with your self-hosted instance."

As noted above, the server where I'm hosting Plausible does not have a dns entry / isn't part of a domain.  AM I able to specify it's IP instead of "the domain name where you installed it "?

Thanks for the smtp workaround, that got me into my self-hosted instance.  Although I'm not getting any traffic into it yet for some reason (no errors shown, either in the Wordpress plugin or at the self-hosted instance of Plausible)

Cheers!

**Reply by Philipinho** [reply]
SMTP configuration is required to send emails from your server.
You can also use a third party SMTP client. 

More info here: https://plausible.io/docs/self-hosting-configuration#mailersmtp-setup

To answer your domain question. 
Yes, Plausible can work with an IP address.

**Reply by steveharman** [reply]
> SMTP configuration is required to send emails from your server.
> You can also use a third party SMTP client.

Sure, but it doesn't _say that_ in the Plausible instructions and sending emails wasn't really an expectation of an analytics tool, so it may not cross people's mind that having SMTP setup is a requirement of their server....unless Plausible says so.

Anyway I'm in now but don't see any data being sent from my site to my self-hosted instance.  There are no errors at either side (the Wordpress webserver or the self-hosted Plausible instance).

Any idea where I can look for debug data or logs please?

Thanks

**Reply by Philipinho** [reply]
Is it stuck on "Waiting for first pageview
on xyz.com" ?

**Comment by steveharman** [comment]
Correct, it is indeed!  :-)

![plausible](https://user-images.githubusercontent.com/12557929/110352190-88af3700-802d-11eb-84da-9bbf7684e16b.png)

**Reply by Philipinho** [reply]
The browser is blocking the request because it is not https.
I.e mixed content.

You have two options:
1. Hence letsencrypt does not support IPs, you'll have to purchase an SSL for the IP address.
Edit: zerossl.com is free and supports IP address. 

2. Create a subdomain and point it to the IP address. 
Use a reverse proxy and then certbot to issue SSL for the subdomain.

I hope this helps.

**Reply by steveharman** [reply]
HI,

The server where I have the plausible Docker instance is now secured by an LE certificate, served by HAProxy as its SSL termination.  I can now connect securely to the instance via https in the browser.  But it still isn't working. I just get  a pulsating green dot i"Waiting for first pageview".  

I have the Plausible Wordpress plugin installed on my website and have put the URL of the server where my Plausible instance is running.  I'm a little confused by the wording in the plugin "...enter the domain name where you installed it " - the _domain name_, or the server's _hostname_?  i.e;  `mydomain.com` or `host.mydomain.com`

I've tried both without success.  Also stopping & starting the Docker instance and reinstalling the Wordpress plugin and accessing my instance in incognito / private mode in case an extension was getting in the way

Any ideas would be very welcome.

Thanks!


<img width="1070" alt="secured" src="https://user-images.githubusercontent.com/12557929/110499181-e5732600-80ef-11eb-8bb5-932ea6eb1d37.png">

**Reply by Philipinho** [reply]
Access the WordPress site where the tracking code is installed. 

`Ctrl + Shift + I `  to inspect element.
Select the `Network` tab.
Refresh the page.

Check the `plausible.js` network requests and see what it says.

**Reply by couetilc** [reply]
I've installed plausible on one of my subdomains, and for the plausible BASE_URL configuration value (as well as virtual.host for caddy's configuration), I entered the subdomain NOT the root domain.

Have you looked at the network logs when your wordpress site loads? The analytics script from your plausible installation will send a `POST /api/event` to your plausible host, perhaps that is getting blocked or not being sent to the correct server?

(also I don't think this is your issue, but at first, it wasn't working for me either. Then I notice in plausible's logs that BASE_URL needs `https://` or `http://` in the url.  So, for example, if I am hosting plausible at `stats.mydomain.com` using ssl, then BASE_URL should be `https://stats.mydomain.com`)

**Reply by steveharman** [reply]
> Access the WordPress site where the tracking code is installed.
> 
> `Ctrl + Shift + I ` to inspect element.
> Select the `Network` tab.
> Refresh the page.
> 
> Check the `plausible.js` network requests and see what it says.

HI,

Well this is very odd.  

Whilst poking about in the browser's dev tools and seeing everything coming back 200 - on my other screen I noticed a graph appear where once there was a pulsating green dot!  (i.e;  my plausible instance seems to be receiving data)

atm @couetilc  I have just the hostname of the instance configured in the Wordpress plugin without `http` or `https` and it appears to be functional, I'll try it with https as you suggest.  

The wording of the plugin is a little vague in the Self Hosted tab; "enter the domain name" - I'm pretty sure it's asking for the hostname where the instance is, not the raw domain name.

Great stuff though, let the analysing commence.
